### ISSUE TITLE
Define "prover" in addition to "holder".

### ISSUE THREAD

**Issue by OR13** [w3c/vc-data-model#902]
Everyone with access to plaintext is a holder... so the term provides very little value.

In protocols, very often the `issuer` is the first holder, and then they `present` to a second `holder` who is often a `subject`.

This second `holder` will then `present` to a `verifier`.

- Issuers issue credentials.
- Provers present credentials using presentations.
- Verifiers verify presentations.
- Holders are anyone who sees a credential.

Holder lacks implied directionality, in practice it has been used to root the direction arrow between a subject and a verifier.

Holder: <-> 
Prover  (sender): -> 
Verifier (receiver): <-

The lack of directionality in the term holder makes it a dangerous term for protocols to rely on, unless the protocol is really intending for no directionality to be implied... which is not the case with the term today, since its only defined in the context of Verifiable Presentations which has a directionality towards a verifier... from a holder.

**Comment by dlongley** [comment]
`Presenter` (instead of `prover`) may provide a wider scope to capture more use cases and better match with the fact that they are "presenting".

**Comment by agropper** [comment]
Requester (instead of holder or prover) is the entity that is making a
request of a verifier. Verifications hardly ever stand alone. I the general
case, the presentation of a credential is part of a request:
- Resource or processing action desired
- Purpose of the request
- Credentials associated with the request
- Payment or refundable deposit to compensate for the significant cost of
policy evaluation and verification.

Obviously, interoperability at the level of requests is complicated and
domain-specific. Formulating the request has domain-specific actions
(resource, purpose, and most credentials) as well as generic actions
(signing and payment). This complexity promotes platforms and
centralization as the client sophistication and policy evaluation are
"sponsored" through surveillance capitalism and platform network effects.

Even if requests are out of scope for this WG, using Requester will help
other efforts.

**Comment by TallTed** [comment]
`Prover` suggests to me that they are playing a specific role in the cryptographic layers, which they are not in any of the scenarios discussed in this issue (including those in comments), so I think that label would cause more confusion than it may relieve.

Similarly, `requester` feels like it inverts the role typically being played by an entity offering a VC/VP for verification.

`Presenter` seems to be the least likely to cause more confusion than it resolves.

I think further discussion along these lines would be aided by reference to (and possibly generate valuable feedback on) the mermaid diagrams that have recently been of significant help in clarifying the VC API Use Case efforts (start [here](https://pr-preview.s3.amazonaws.com/eric-schuh/vc-api-use-cases/pull/19.html#get-digital-permanent-resident-card) and scroll down a bit....).

**Comment by vongohren** [comment]
We have touched upon similar problems and ended up with owner. But I feel presenter might be a better option.
We did use receiver internally, but verifier can be just as good

**Comment by agropper** [comment]
Owner is an interesting choice. It accurately reflects the reality that an
owner can delete or take the VC off-line regardless of the subject or
contents. It also matches the common use or Resource Owner in OAuth, GNAP,
etc...

Presenter, Holder, and Requester are different from Owner in terms of
delegation. It makes little sense to delegate ownership and leads to
impersonation and lack of accountability as major security flaws.

As one who argues for freedom to delegate as a human right, I would suggest
we avoid Owner.

**Comment by brentzundel** [comment]
If presenter is the party who submits a presentation to the verifier, what is the name of the party that receives a credential from an issuer?
My understanding was that holder (as a role) was both the party who receives the credential from an issuer and presents a proof to a verifier.

The most recent spec defines it this way: 

A role an [entity](https://www.w3.org/TR/vc-data-model/#dfn-entities) might perform by possessing one or more [verifiable credentials](https://www.w3.org/TR/vc-data-model/#dfn-verifiable-credentials) and generating [presentations](https://www.w3.org/TR/vc-data-model/#dfn-presentations) from them. A holder is usually, but not always, a [subject](https://www.w3.org/TR/vc-data-model/#dfn-subjects) of the [verifiable credentials](https://www.w3.org/TR/vc-data-model/#dfn-verifiable-credentials) they are holding. Holders store their [credentials](https://www.w3.org/TR/vc-data-model/#dfn-credential) in [credential repositories](https://www.w3.org/TR/vc-data-model/#dfn-credential-repository).

**Comment by OR13** [comment]
presenter: The party A, such that A presents to B using a verifiable presentation.
receiver: The party B, such that B verifies a verifiable presentation form A.
holder: an ambiguous term that applies to both presenters and receivers, and conveys no directionality

**Comment by jandrieu** [comment]
I'd like to suggest that we split "Holder" into "Recipient" and "Presenter", with the explicit understanding that it is generally assumed that these three are all the same person.

**Comment by OR13** [comment]
For the record:

PROPOSAL: define holder (undirected), receive / present (directed actions) .... receiver, presenter (directed).

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-09-07-vcwg#section2-3) on 2022-09-07

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.3. Define "prover" in addition to "holder". (issue vc-data-model#902)

_See github issue [vc-data-model#902](https://github.com/w3c/vc-data-model/issues/902)._




**Orie Steele:** We created the spec to have a term for presenting.  
… It's important to acknowledge the different roles for being a holder.  
… It's dangerous to have terms in VPs that are undirected. The holder label is undirected.  
… It would be better for the entity making the presentation to be able to assert that "I am presenting or I am receiving".  
… We want directed intentionality in the data model.  
… That's why we need additional terms related to the holder role.  

**Kristina Yasuda:** There's a suggestion to use presenter.  

**Joe Andrieu:** It makes sense to add both recipient and presenter for what we're currently calling holder.  

> *Manu Sporny:* +1 to defining the roles in a directional manner.

**Joe Andrieu:** That's an iteration on this conversation.  

**Kristina Yasuda:** There is a consensus to have two terms.  

**Manu Sporny:** Are we talking about getting rid of holder or saying that holders have two things that they can do.  

> *Orie Steele:* +1 to the latter formulation manu.

**Manu Sporny:** I wondering if issuers can receive and present as well.  
… It's a question we have to answer for each of the current roles we have.  

**Joe Andrieu:** We would benefit from keeping holder and clarifying it.  

> *Brent Zundel:* +1 Joe.

**Joe Andrieu:** The issuer/holder/verifier is a very powerful group of terms.  
… We should elaborate on them.  

> *Joe Andrieu:* +1 to chat at TPAC.

**Kristina Yasuda:** This is the first time we're talking about this issue.  
… Let's give people time to think about it.  

</details>

**Comment by agropper** [comment]
The role of holder is the same as requester. An entity requests a VC from an issuer and may provide scope as well as authentication as part of the request. (Another) entity requests authorization from a verifier and may provide scope and purpose as well as authentication as part of the request. Unlike holder, requester is always a directed term.

**Comment by jandrieu** [comment]
So, now you do want to bikeshed?

**Comment by agropper** [comment]
It's not bike shedding to consider a directed request-based model vs. the current discussion of undirected and directed.

**Comment by logpo** [comment]
Splitting the holder definition in two will create more confusion IMO. The holder holds the credential and the means to present that credential to a verify. So they receive credentials from issuers, but they have a private key which is what separates them from anyone else who has the plaintext credential. I agree that the term isn't perfect but I'm not sure splitting the role in two fixes that.

**Comment by David-Chadwick** [comment]
This problem of describing the actions is compounded when subject NE holder. Describing the holder's relationship with the issuer (and subject) in this case is complex due to the different relationships there may be, and the holder may have to proove possession to the Verifier when the VC is not a bearer VC.
When subject EQ holder we have the public key in the VC to prove the relationship/possession.
If we mirrored this for issuing VCs to holders (NE subject) it would simplify these relationships, because the VC would contain the holder's public key and then PoP to the verifier becomes straightforward. Of course it does not deal with the subject EQ holder passing their VC to another holder (NE subject). But that is a second order issue.

**Comment by decentralgabe** [comment]
I'd like to re-define `holder` as one who has the cryptographic ability to prove control of a credential **using a cryptographic identifier or key defined by the issuer of the credential**.

**Comment by OR13** [comment]
I'd like to deprecate `holder`, and NOT define it normatively, instead, I would like to define only terms that have some security characteristic associated with the securing format (vp-jwt, vp-data-integrity).

Unsigned "verifiable" presentations are the only use case for the existing `holder` term imo.

**Comment by agropper** [comment]
There are significant human rights associations with the `holder` term because of the temptation to use a biometric wallet or, more literally, an ankle bracelet. 

"Client credentials" is a constant challenge. To the extent that `holder` is a client, the relationship of the client to the human that is operating the client must be **explicit** in our normative statements. The risk of being expected, if not forced, to carry or use a certified client is an obvious human rights problem.

**Comment by TallTed** [comment]
The `holder` *holds* the VC.  They need not be identified anywhere in the VC (though they might be).  There is no global need to tie VCs to their `holders`, though it may be useful in some cases.  Likewise, there is no global danger for `holders` of VCs, nor even `subjects`, who may not ever know they were the `subject` of a given VC.

I am growing increasingly weary of saying the same thing again and again, of having to defend against universals that *aren't* universal, certainly technologically and generally otherwise.

Please remember, we are working on the _technological_ aspects.  Concerns that amount to "government logic", akin to business logic, are pretty far outside of our scope, unless a specific harm can be identified and defined and demonstrated to be inherent to the tech, which will enable us to construct a defense against that harm, at least to the extent that we agree about its significance.

**Comment by RieksJ** [comment]
One of the above comments nicely distinguished between parties and roles. A [party](https://essif-lab.github.io/framework/docs/essifLab-glossary#party), whereas a role can be seen as a particular way in which it can perform. It's like a play where actors fulfill roles, which is easily transferrable to the real world (I can act in a role as 'father', 'TNO employee', etc.). The VC-image of issuer-holder-verifier is a bit misleading in the sense that many seem to interpret these roles as if they were the party itself - as little children do when they see Santa (for the Americans) or Sinterklaas (for the Dutch).

Having said this, we just need to identify the various 'parts' of the play, e.g. 'issuing', 'requesting/receiving' - whatever we choose/need to have, then properly describe these 'parts' and define a name by which we can refer to parties that will perform this part/assume this role. It doesn't really matter what the role-name is, as long as it is properly defined. But of course, it helps if these names trigger the right 'hallucinations' with readers.

**Comment by OR13** [comment]
A related issue discussed on the traceability call today:

- https://github.com/w3c-ccg/traceability-vocab/issues/521

**Comment by awoie** [comment]
I also found the term holder a bit vague. We could try to run a non-binding poll on this and ask folks how they feel about introducing a new role `presenter` (I also feel that prover is too narrow in scope since it is not always about verification of cryptographic proofs).

- Issuer issues VCs
- Verifier verifies VPs/VCs
- Presenter presents VPs/VCs

I'm ok with keeping the term Holder in the VCDM as non-normative text if people think this might help with establishing some reader context but if we agree on presenter, then we should probably make `presenter` a normative property in the VP.

**Comment by awoie** [comment]
Also note that `holder` has never been normatively defined in VCDM 1.1.

**Comment by OR13** [comment]
It's amazing how many critical issues there are for us to work on.

My guess is that "holder binding" would be far less contentious if we could define what a holder is first.

My sense is that "holder binding" might more accurately be titled "presenter binding".

**Comment by TallTed** [comment]
I continue to believe that "binding" is the wrong word for this association, this desired _restriction_ (which *might* be the right word for it).

**Comment by OR13** [comment]
Great point about the word binding, it makes you wonder if we are trying to relate "holder binding" to "token binding" or if that relationship is not what we are hoping to accomplish:

https://www.w3.org/TR/webauthn-2/#dom-collectedclientdata-tokenbinding

**Comment by dlongley** [comment]
@OR13,

I think the "holding binding" issue(s) and related conversations started from the idea of creating a dependency of sorts on a proof of possession, but it's grown to encompass more than just that idea. Besides, that idea had some initial flaws when called "binding" that people mentioned early on as well.

I agree with @TallTed that we need a better word than "binding" -- but I'm hoping at this point that we don't try to bikeshed it but rather let something new come out of further discussion. I think there are a number of different use cases people are interested in that all fall under the umbrella of "holder binding". Some examples that come to mind:

1. Having an issuer say that they intended a particular entity to be the presenter of a VC (but in the absence of why that matters?).
2. Having an issuer say that verifiers could examine certain evidence or compare it against the presenter of a VC (perhaps running checks X, Y, Z against them) to see if they are the intended subject of the VC / some other identified and intended party.
3. Having the issuer say that they believe a particular party should be given some privilege and perhaps suggest how to authenticate that party. Note that maybe it doesn't actually matter who *presents* the VC unless the act of presenting involves sufficient authentication of said party. (There's also some tension here between what the verifier might accept as valid authentication and what the issuer might prefer or suggest they accept).

**Comment by agropper** [comment]
This ongoing discussion has not converged for as many years as I've been
around. If anything, it seems to be getting fuzzier even as ACDC and
other options come on the scene. The challenge from ISO mDL and OWF is also
adding to the fuzz.

I think some of this is due to not treating biometrics as primary
components of VCs. To help advance our discussion, I posted a slightly
tongue-in-cheek perspective on digital human identifiers to CCG today.
https://lists.w3.org/Archives/Public/public-credentials/2022Nov/0195.html

It's not proposing a solution, but hoping people take the question
seriously and factor it into the VC 2 design.

Adrian

**Comment by andrewhughes3000** [comment]
At ISO SC17 WG4 For 23220-4 we are discussing the same thing. 
We are trying to determine whether there is value in distinguishing between the human entity whom the issuer identifies before issuance, the main subject of a credential, the entity receiving a credential directly from the issuer, the authorized user of a wallet, a human guardian of the subject, a delegated person, an entity authorized to assemble a presentation, an entity authorized to present a presentation, a prover. I might have missed some. 
We are going down the path of having the issuer endorse a "means to authenticate" which, when a verifier wants confirmation that the entity presenting a credential is authorized to do so, can be invoked by the <something> which then releases a signal to the verifier that it is indeed the case. 
Needless to say we have a long debate ahead of us - and we have the benefit of needing to involve the issuer in the mix instead of having no guardrails.

**Comment by TallTed** [comment]
> the benefit of needing to involve the issuer in the mix

That's what's known as "phoning home," which has significant privacy implications.

**Comment by andrewhughes3000** [comment]
Nope - that's not what I'm referring to. I mean that we are looking at how the issuer could pre-approve the means to confirm the "means to authenticate". So that it's not completely up to the wallet to decide how to do it. It constrains the range of options we must consider. 
You have to believe me when I continue to say that almost all "mDL People" dislike the "phone home" server retrieval modes. And believe it or not, all of us are very concerned about building privacy-preserving capabilities into the protocols.

**Comment by RieksJ** [comment]
@OR13 said
 
> My guess is that "holder binding" would be far less contentious if we could define what a holder is first.
> 
> My sense is that "holder binding" might more accurately be titled "presenter binding".

My guess is that we could resolve all this by reverting to some basics.

1. **First understand the generic problem, then come up with solutions, next choose which (not) to standardize**. I haven't seen this so far. I have seen lots of hand-waving, lots of examples that are specific occurrences of some presumed generic problem, but no clear, simple description of the generic problem that we're trying to solve, and which of the roles (issuer/holder/verifier/presenter/...) actually suffers from that problem.
 
2. **The semantics of every identifier that an issuer uses to identify the subject of a claim MUST be controlled by the issuer of (the credential that contains) that claim**. By 'identifier semantics', I mean the mapping between the identifier and the entity that this identifier refers to. In other words, issuers MUST be the authority for dereferencing any subject identifier in a claim they author. If they do not, then they may sign a claim stating e.g., that `did:ex:alice` has earned some degree assuming that `did:ex:alice` refers to Alice, but as they do not control the semantics of `did:ex:alice`, this identifier may at some later point in time point to Bob, and a verifier may incorrectly assume that Bob has earned that degree (and perhaps sue the issuer for malpractice).

3. **Since the issuer MUST be the authority that controls the identifier semantics of every identifier in a VC, it is also the party that MUST authoritatively determine which mechanisms (if any) other parties - specifically: verifiers - can use to learn what entity that identifier refers to**. This then also holds for all subject identifiers in the claims of a VC.  My sense is that issuers could add such mechanisms as part of the claims, e.g., as suggested by #760, which holders could selectively disclose to verifiers as they see fit.

My sense is that verifiers (or others) do not need to find out whether or not the holder/presenter/... is the subject of a claim, but rather that (the generic problem is that) they need to do the converse, i.e, establish which party (or other entity) is being referred to by a subject identifier. I propose that #760 has one of the solutions, and I would like to see others so we can then choose what to (not) standardize.

**Comment by David-Chadwick** [comment]
I would rename 1. as "First define the roles in the system" There are 4 roles involved, not 3 as Orie suggested. The obvious one that he has missed is the issuee role.
Issuer issues VCs
Issuee receives these VCs 
Verifier verifies VPs/VCs
Presenter presents VPs/VCs

You will note that the holder role never appears in any of these 4 roles , because it is a role that is not visible outside of the wallet. i.e. whenever the wallet receives or transfers a VC or VP, it is acting in one of the above 4 roles, and is never acting as a holder. So we need to downplay or even remove the role of holder.

2. is interesting, but not sufficient. I would say that the Issuer needs to identify every role that is pertinent to the VC that it is issuing. So, an issuer could if it wishes, identify the verifier that a particular VC should be given to.

**Comment by RieksJ** [comment]
@David-Chadwick I disagree that 1. should be renamed. I can see that you need to define some roles (and perhaps some other terminology) in order to state the generic problem but skipping the formulation of the generic problem is a no-go for me.

**Comment by bumblefudge** [comment]
Big thanks to @andrewhughes3000 for lending a nuanced vocabulary!

Abbreviation|Andrew's description
---|---
A|the human entity whom the issuer identifies before issuance
B|the main subject of a credential
C|the entity receiving a credential directly from the issuer
D|the authorized user of a wallet
E|a human guardian of the subject
F|a delegated person
G|an entity authorized to assemble a presentation
H|an entity authorized to present a presentation
I| a prover

Just for the sake of spitballing, all the following statements feel reasonable to me:
1. A and B are assumed coequal by today's spec or at least my mental model of it
2. A == B == C == D in a lot of use cases and there may be little consensus or interest in discussing those use-cases where this is not the case! The [Verite](https://verite.id) credentials we've been modeling, for instance, would be considered illegal (i.e. invalidated extra-technically, warranty voided, etc) if not, or if D were a superset of A/B/C.
3. E and/or F and/or G would somehow need to be constrained at time of issuance for a "presenter-constraint"/"holder-binding"/etc algorithm to make sense.
4. G<>H<>I feel to me like distinct functions of the mechanics of presentation that might be done by different instances/tenants/etc in large-scale prod (in the same sense that VC API slices up roles into functions). It might make sense for constraints on delegation or distributed proofing functions to get this granular in some cases, and not to do so in others.  Implementation detail?

Feel free to disagree or ignore, I just found it helpful to type out my thoughts.

**Comment by David-Chadwick** [comment]
@RieksJ The generic problem from my perspective is this. How can we replace the dozens of plastic cards, paper-based and other physical credentials that we carry around with us and/or continually have to show to others, with the electronic equivalent of them. Do you agree?

**Comment by David-Chadwick** [comment]
@bumblefudge A == B == C == D will cater for the vast majority of use cases (most probably >>90%). It's the other minority of use cases that present the most complex problems to be solved and will take >>90% of our effort.

**Comment by bumblefudge** [comment]
> @bumblefudge A == B == C == D will cater for the vast majority of use cases (most probably >>90%). It's the other minority of use cases that present the most complex problems to be solved and will take >>90% of our effort.

Well if that sentiment has consensus we can narrow in on a much more manageable scope for the E-thru-I constraints and proofs, at least for the purposes of this issue 😅

**Comment by TallTed** [comment]
> [@andrewhughes3000] You have to believe me when I continue to say that almost all "mDL People" dislike the "phone home" server retrieval modes. And believe it or not, all of us are very concerned about building privacy-preserving capabilities into the protocols.

That's great. Your previous comment about _needing to involve the issuer in the [verification] mix_ suggested (and suggests) otherwise. Your later statement that _the issuer could pre-approve the means to confirm the "means to authenticate"_ does help to ease the "phone home" concern somewhat — but I hope you agree that these statements are very different, and can see why I was concerned by the earlier one.

We need to take great care in these discussions, to express exactly what we mean in a manner that is least likely to be misunderstood. This may make many comments far more verbose, but that clarity is vital if we're to reach anything approaching consensus across the various related projects, nations, and organizations involved.

**Comment by TallTed** [comment]
> [@David-Chadwick] A == B == C == D will cater for the vast majority of use cases (most probably >>90%)

What tools have you used, and how broad a population did you survey, to arrive at this 90% assertion?

If, as I suspect, you simply plucked it from the air, please stop asserting it. Such evidence-free assertions do no-one any favors.

**Comment by TallTed** [comment]
Also, everyone, please note that a system designed to handle situations where each and every role (however many that may be) is filled by a different entity (e.g., `A != B != C != D`) may easily be used for situations where any number of those roles are filled by a single entity (e.g., `A == B == C == D`), but the opposite is not true.

Thus, given that we know of situations where each and every role is filled by a different entity, we SHOULD develop a solution which serves for this situation. Failing that, we MUST develop a solution which can be adapted to serve this situation in the future, without breaking any deployments that may exist at that time.

**Comment by RieksJ** [comment]
@OR13 wrote

> My guess is that "holder binding" would be far less contentious if we could define what a holder is first.

VCDM [defines the term](https://www.w3.org/TR/vc-data-model/#dfn-holder), so that is NOT the problem. I do see some others.

1. **terms that have a definition are not used in the meaning as defined**, which causes misinterpretations and lots of needless disputes about these terms that do not get resolved (which I conclude from the fact that they keep popping up). (AnonCreds seem to do this differently, see e.g., their [issue](https://github.com/hyperledger/anoncreds-spec/issues/100) on ambiguous use of their term `holder`)

2. **terms are introduced of which their meaning is assumed to be self-evident** - i.e.: without providing a definition that actually allows others (e.g., non-native english speakers) to determine what is, and what is not an instance (example) of the term. Although the cause is different (the definition lacks), the effect is very similar, as terms are used without a proper reference to their meaning.

3. **ignoring the fact that terms have a scope/context in which they have a particular meaning** gives rise to misunderstandings. For example, in the context of the Dutch government, the `holder` (of an identity document) is [defined](http://wetten.overheid.nl/jci1.3:c:BWBR0005212&hoofdstuk=I&paragraaf=1&artikel=1) as "the person in whose name the travel document is issued and for whom it has been issued". In VCDM terminology, that person would be referred to as the subject (of the identity document), and the holder would be the entity that possesses it and can present it, which could be, but is not necessarily its subject.

Assuming that all issues in this repo should be relevant for (the context of) VCDM, I propose that we make a real effort to use the terms that are defined ONLY in the meaning as defined in the VCDM. If additional terms are needed (or changes in existing ones), a proposal to do so should be raised in which a definition is suggested that allows people that work in the VCDM context to agree on what does, and what does not, constitute an instance/example of that term. I'm curious to see how this would affect the discussion on 'holder binding'...

**Comment by RieksJ** [comment]
@David-Chadwick 

> @RieksJ The generic problem from my perspective is this. How can we replace the dozens of plastic cards, paper-based and other physical credentials that we carry around with us and/or continually have to show to others, with the electronic equivalent of them. Do you agree?

Yes and No. Yes, because I cannot argue what your perspective is or isn't. No, because we should not _replace_ all that, because many people wouldn't be able to actually use wallets. It should be an alternative that people can use. 

But mainly 'No' because what I'm really looking for is a convincing and understandable argument for proposing the term "prover" (in addition to "holder") that I as a (non-code writing) member of VCC can at least understand. I have no clue what phrases such as stuff like "(implied) directionality", or "the direction arrow" refer to (they are not in the VCDM). I cannot make any sense of sentences such as "Holder lacks implied directionality, in practice it has been used to root the direction arrow between a subject and a verifier" and "The lack of directionality in the term holder makes it a dangerous term for protocols to rely on", as @OR13 wrote in the initial post. So how could I contribute?

If the problem is only an implementation issue, there is no need to introduce the term "prover" in the VCDM. If the problem is generic, i.e., relevant in the context of VCDM, I think I should be enabled to understand the problem that @OR13 attempts to address in this issue.

**Comment by OR13** [comment]
Not clear what the action is here, seems blocked by holder binding debate.

**Comment by RieksJ** [comment]
The original issue by @OR13 is a terminology issue that says: he guys, the the term 'holder' is about a single role that is related to two different activities. First, it is a role that "an [entity](https://www.w3.org/TR/vc-data-model/#dfn-entities) might perform by possessing one or more [verifiable credentials](https://www.w3.org/TR/vc-data-model/#dfn-verifiable-credentials)". Second, it is a role that "an [entity](https://www.w3.org/TR/vc-data-model/#dfn-entities) might perform by [...] generating [presentations](https://www.w3.org/TR/vc-data-model/#dfn-presentations) from them". Since these activities are distinct, it makes sense to make these roles distinct as well, and have different names for them.

Concrete action: change the VCDM spec as follows:
- split the currently defined holder role into two parts, the first one being the "possessing one or more [verifiable credentials](https://www.w3.org/TR/vc-data-model/#dfn-verifiable-credentials)" and the other one being "generating [presentations](https://www.w3.org/TR/vc-data-model/#dfn-presentations) from them", and choose names for them (e.g. "holder", and "presenter").
- go through the VCDM text and replace the occurrences of 'holder' with 'presenter' wherever that is appropriate.

I would suggest to also add some text that clarifies the term 'role', inspiration for which is [here](https://essif-lab.github.io/framework/docs/essifLab-glossary#role) and [there](https://essif-lab.github.io/framework/docs/essifLab-glossary#role-name).

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-02-08-vcwg#section4-2) on 2023-02-08

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 4.2. Define "prover" in addition to "holder". (issue vc-data-model#902)

_See github issue [vc-data-model#902](https://github.com/w3c/vc-data-model/issues/902)._




**Brent Zundel:** define prover in addition to holder. What to folks think?.  
… not sure if we are at a point of agreement on next steps.  

**Orie Steele:** issue describes itself. Main problem is association with holder binding.  
… become inactionable due to this.  
… original intention was to describe that holder undirected. Might want to make it clear the direction.  
… suggest entire data model blocked by holder binding conversation.  
… don't have def with requirements at presentation layer that is useful.  
… hopefully untangle at f2f.  

**Manu Sporny:** orie took shot at high level definitions, longley added some suggestions. Sparked massive discussions.  
… could be that we don't need yet another role or term in ecosystem.  

**David Chadwick:** I think we need two new terms. Get rid of holder all together.  
… two new terms issuee and presenter/prover.  
… those roles never change.  
… holder always changes because vcs passed between holders.  

> *Juan Caballero:* +1.

**David Chadwick:** holder does not participate in anything particularly.  

> *Phillip Long:* +1 to David C's statement about holders as indeterminate, not having a defined actionable role..

> *Juan Caballero:* good point phil-- yes, +1 to that :D.

**Brent Zundel:** thanks for convo. moving on.  

</details>

**Comment by mwherman2000** [comment]
> Since these activities are distinct, it makes sense to make these roles distinct as well, and have different names for them.

@RieksJ Here's another approach: https://youtu.be/uo4RuT__IXw?t=575s

**Comment by Sakurann** [comment]
having read the thread, I would suggest proceed as following: 
- first discuss whether to break down what is currently one entity "holder" into multiple entities
- if yes, what would be the break down? suggestions seem to be three entities: an entity that receives, an entity that holds, and an entity that presents a VC. what are the terms/definitions for the new terms/
- if no, is there a need to normatively define "holder"

**Comment by agropper** [comment]
We might consider:

   - Adopting the Terminology from the RWOT11 paper
   https://github.com/WebOfTrustInfo/rwot11-the-hague/blob/master/final-documents/identifier-binding.pdf

then

   - Discuss the choice of a VC-centric vs. a User-centric approach. Make
   it clear why we picked one over the other.

If we choose a user-centric approach then all VC interactions look like a
Request-Response where an entity either:
- requests a VC based on presenting claims including other VCs or
- requests a service based on presenting claims including a VC

- Adrian

**Comment by David-Chadwick** [comment]
> "break down what is currently one entity "holder" into multiple entities"
>
I am strongly in favour of this. But I think that only two entities are needed, the entity that receives the VC from the issuer and the entity that presents the VC to a verifier. Any entity that either receives or presents a VC must be one of these two entities (or the issuer or verifier). Therefore the holder is never actually visible as a separate entity so we can dispense with this role

**Comment by RieksJ** [comment]
@David-Chadwick I agree that we need two roles to replace the current holder role. However, since VCs can be transferred in their own right, that is: not as part of a VP, we may need to decide how to refer to the role of an entity that doesn't issue a VC, but simply passes it on as is. 

There are many use-cases for this: a simple one is where I, as an employee of the organization TNO, may want to pass on a credential with my name and address to TNO's traveling department so that they can book a flight with me being the passenger, which is the SSI parallel of me handing (a copy of the personal page in) my passport to that travel department (which I trust to not do anything else with it). This is very much in line with the idea that is often voiced by @jandrieu that anyone can hold VCs (that contain claims) about anyone/anything else.

**Comment by David-Chadwick** [comment]
Hi Rieks
    
    On 14/02/2023 19:27, Rieks wrote:
    
    
      
      @David-Chadwick I agree that we
        need two roles to replace the current holder role. However,
        since VCs can be transferred in their own right, that is: not as
        part of a VP, we may need to decide how to refer to the role of
        an entity that doesn't issue a VC, but simply passes it on as
        is.
    
    I completely accept the fact that anyone can hold any VC. That is
    the VC model. The real issue the verifier has to verify, is: is the
    holder entitled to hold and present this VC for the transaction in
    hand, or did he just get it from somewhere, in which case the VC
    should not legitimately be used for the current transaction.
    
      There are many use-cases for this: a simple one is
        where I, as an employee of the organization TNO, may want to
        pass on a credential with my name and address to TNO's traveling
        department so that they can book a flight with me being the
        passenger, 
    
    Actually I don't think this is a realistic use case. My admin has
      often bought tickets for me and I have never given then a copy of
      my passport for them to do this. They know who I am as my details
      are in their database. Futhermore no airline requires to see a
      passport in order to sell anyone a ticket. I could buy an airline
      ticket for you today.
    
    Kind regards
    David
    
    
      which is the SSI parallel of me handing (a copy of
        the personal page in) my passport to that travel department
        (which I trust to not do anything else with it). This is very
        much in line with the idea that is often voiced by @jandrieu
        that anyone can hold VCs (that contain claims) about
        anyone/anything else.
      —
        Reply to this email directly, view it on GitHub, or unsubscribe.
        You are receiving this because you were mentioned.Message
          ID: ***@***.***>
      [
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/w3c/vc-data-model/issues/902#issuecomment-1429190718",
"url": "https://github.com/w3c/vc-data-model/issues/902#issuecomment-1429190718",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by mwherman2000** [comment]
> > Since these activities are distinct, it makes sense to make these roles distinct as well, and have different names for them.
> 
> @RieksJ Here's another approach: https://youtu.be/uo4RuT__IXw?t=575s
> 
> You may need to watch the following as background: https://youtu.be/uo4RuT__IXw?t=401s

In case you didn't have time to watch these video snippets: There is a generalization of the Issuer-Holder-Verifier model that separates the Holder role into two separate roles:
1. The Holder as a specialization of the Credential Receiver role (receiving VCs from an Issuer - the latter being a different specialization of the Credential Sender role)
2. The Holder as a specialization of the Credential Sender role (sending VPs or VCs to a Verifier - a later being different specialization of the Credential Receiver role)

Here's an example diagram...
![image](https://user-images.githubusercontent.com/6101736/218681819-3bf8cd8d-311d-49bc-acc9-5cdb52b7626b.png)

Here's a example diagram of the generalized Credential Sender-Receiver model (which I think should be documented first in the spec to faciliate the explanation of the Issuer-Holder-Verifier model)...
![image](https://user-images.githubusercontent.com/6101736/218682206-d2256f44-b45d-4df8-818a-61de3b0da99b.png)

**Comment by RieksJ** [comment]
@David-Chadwick wrote

> The real issue the verifier has to verify, is: is the holder entitled to hold and present this VC for the transaction in hand, or did he just get it from somewhere, in which case the VC should not legitimately be used for the current transaction.

As a verifier, I have no clue how to determine whether or not the holder is entitled to hold and present a VP/VC/claim. Who provides such entitlements? Who enforces them? 

From the perspective of a verifier, would an arbitrary person not be entitled to hold/present a VP/VC/claim about e.g. the right of a child to travel (by train, in an airplane, ...)? All the verifier cares about is whether or not the subject of that claim can be related to that child, regardless of who presents it.

I think the real issue is that verifiers need ways to determine who/what the subject of an identifier is (and by extension: of a claim). *If* the verifier has such a capability AND it chooses (for specific use-cases) to enforce a business-rule that states that it will only accept a claim when it is provided by its subject, it can easily do so.

**Comment by agropper** [comment]
This brings us back around to a long conversation almost two years ago in a
thread labeled “
*VC-HTTP-API - A follow up on the RAR presentation”*

It’s the “Cruise-Ship Use Case” where Alice trusts the cruise line and the
cruise line outsources verification and validation of Alice’s credentials
to an expert that Alice cares nothing about.

Adrian

**Comment by David-Chadwick** [comment]
> As a verifier, I have no clue how to determine whether or not the holder is entitled to hold and present a VP/VC/claim. Who provides such entitlements? Who enforces them?

I agree. The essence of my proposal is that the holder provides hints to the verifier to help the verifier, and these hints point to the various VCs in the VP, indicating how the verifier can verify that the holder is entitled to hold each VC. The slides are in the deck of the F2F meeting that was held today. See 
https://docs.google.com/presentation/d/128DHWSzVxPgAhB0mq-h23_iATnbVeA4Y-JhNLjpcXJE/edit#slide=id.g1f24e2c0aad_7_602
starting at slide 48

**Comment by RieksJ** [comment]
@David-Chadwick: My problem is that I do not understand what it means when you say someone is **entitled** to hold/present a VP/VC/claim. 

An entitlement (a 'right' according to [Merriam-Webster](https://www.merriam-webster.com/dictionary/entitled)) must have a ground, e.g. a law or regulation, an agreement or similar. While there will be specific situations where this might be the case, in the general case there is nothing to ground such an entitlement on.

Thus, can you help me understand what you mean when you say "the verifier can verify that **the holder is entitled** to hold each VC"?

**Comment by David-Chadwick** [comment]
Many VCs are issued to holders (usually subjects) after they have entered into an agreement with the issuer. So this satisfies the definition of entitled I believe. e.g. I am entitled to hold my credit card, I am not entitled to hold your credit card.
Other VCs are deemed to be "bearer" VCs, meaning that anyone/everyone is entitled to hold them. Yet other VCs may be issued to third party holders (who are not the subject) when they enter into an agreement with the issuer. They are also entitled. Does this answer your question about entitlement in general? Now to your specific questions
1.  The dog organisation, being the issuer, determines who may or may not hold the VC on behalf of the dog. I cannot determine this for them.
2. When you got the VC that contains your personal details the issuer will have told you the terms and conditions for the issuing of this VC. So you should read these to see what you are entitled to do with it. E.g. several plastic cards I possess have 'not transferrable' stamped on them. I understood this when I signed the agreement to get the card. This notice makes it pretty clear to everyone who is entitled to hold it. Only the subject is entitled to hold it.
3. Ditto
4. If you look at the use case examples I have given then you will see that there is a deterministic way for the verifier to determine that the holder was indeed entitled to hold the VCs he/she/it is presenting (assuming that none of the VCs that are being presented by someone other than the issuee do not have a ToU that indicates "non transferrable") or that they are bearer VCs where everyone is entitled.

**Comment by RieksJ** [comment]
It is clear to me that by 'entitlements' you mean the rights/duties that are specified in an agreement (contract) that the issuer and the party to which the VC has been issued have committed to. Thank you for that. 

I accept that in the agreement/contract between you and your credit card provider there is a clause that that states that you ma not pass the card around and let someone else use it, and that you are the only one that is entitled to pay with it.

The text you wrote in reply to my specific questions mostly talks about the issuer, the party to which it issues a VC, and an agreement between the two that say what that party could/should (not) do with it, and you refer to earlier stuff to enlighten me of how the verifier could verify such entitlements.

However, you haven't answered questions 2a/b/c and 3a/b/c, that relate to why the verifier would/should care. The way I see it,  rights/duties/entitlements in a contract only have meaning for those that have committed to it (signed it). An arbitrary verifier that didn't do so couldn't care less. When I considered your example with credit cards, I tried to remember of any practical situation where a credit card payment was refused on the mere fact that the presenter was not the 'rightful holder'. I could not find an example of that, and quite a few examples where verifiers do not even bother to check. All they care about is whether or not they will get paid. It is their own business rules that say what assurances they need to get paid, and it is these rules that they seek to comply with. Knowing the card details doesn't make one its rightful holder, but it is all verifiers need to get paid. 

Use-cases in which the verifier has business rules that require that the contents of a claim pertain to the party that presents it do, of course, exist: a verifier that needs to decide whether or not someone gets access to a session in which (s)he can make an exam is one of them. But even in such cases one could often argue that the verifier should allow for the claims to be presented by someone else, e.g. in case the person has some kind of disability. 

I conclude that 
1. most VC would be 'bearer VCs' in the sense that anyone could hold them (which is different from what VCDM says bearer claims are)
2. Verifiers are not interested in whether or not a VC is presented by a 'rightful holder', i.e. a party that has committed to an agreement with the issuer in which its rights/duties (entitlements) regarding the VC are specified. They are only interested in making sure that in the context of the transaction in which it is presented with VCs/VPs, their own business rules are being complied with.
3. whenever issuers put stuff in a VC that informs third parties about the entitlements of a party that the issuer has a contract with, that could be a means that verifiers might use to obtain the assurance (to comply with their own business rules) they need.  But such verifiers would then also need to identify/authenticate the party that had committed to that agreement, and they should not automatically assume that it must be the holder. And that is what the [identifier-binding paper](https://github.com/WebOfTrustInfo/rwot11-the-hague/blob/master/final-documents/identifier-binding.pdf) proposes the means for.

**Comment by David-Chadwick** [comment]
> They are only interested in making sure that in the context of the transaction in which it is presented with VCs/VPs, their own business rules are being complied with.

In which case they must inform the holder what these rules are, since neither the issuer nor the holder in general will know what these rules are. I do not see anything in the holder binding proposal that indicates what these rules are.

Furthermore when the presenter is neither the subject nor the issuee, then holder binding  is not much use since the presenter (who is neither of these) is not able to prove the holder binding without either masquerading as the issuee or subject, or by proving delegation of authority or similar. Neither of these seem to be within scope of the current holder binding.

It seems that in your travel agent use case the verifier does not care who the presenter is, as it is treating the passport as a bearer credential that anyone can present to it, and it will issue the ticket in the name of the passport subject. So holder binding is not needed in this case.

Therefore you have not indicated why. holder binding is actually of any use.

**Comment by RieksJ** [comment]
In practice, verifiers tend not to inform holders (or issuers) what these rules are. Rather, they request the user to supply data (fill in forms, present a VP) that enables them to verify these rules (and because nobody makes a real point of this, data minimization has become a farce).

You seem to insist on thinking that for whatever reason the verifier is interested to learn who the holder (or issuee) is. While I agree that in the majority of the situations that might well be the case, there are also situation where the presenter is neither the subject nor the issuee. In the example of the dog-organization, that organization could issue a VC about the dog to (the partner of) its owner, who could then transfer the VC to friend, asking him to register the dog for a dog show. The registration application (verifier) doesn't care who the friend is, nor who (the partner of) its owner is, and there's no reason to forbid that. This example shows that there are cases where holder binding is irrelevant. 

I argue that holder binding is always irrelevant (in the context of VCDM, that is). What we need is identifier binding, i.e. providing the verifier with the capability to identify/authenticate the entities that identifiers (that are used e.g., in claims) refer to. If delegation of authority or something similar is required, this can be arranged by creating claims that state what the authority is that is being delegated, and state (an identifier for) the party that delegates the authority and (an identifier for) the party to which the authority is delegated. A verifier may be very interested to learn who these parties are that such identifiers refer to. Identifier binding provides a way to enable them to learn that.

I agree that I haven't indicated why holder binding is actually of any use. That is because I think that in the context of VCDM, identifier binding is the more generic problem to be solved, and once we have that, every case in which holder binding would be required would also be solved.

**Comment by David-Chadwick** [comment]
> While I agree that in the majority of the situations that might well be the case, 

And it is these majority of situations that I am addressing. For the other minority, perhaps where the verifier is not interested in who the holder is, this is fine. In its request to the holder, the verifier can indicate that it does not care or can ignore any information that the holder might provide.

> I argue that holder binding is always irrelevant 

I think we can both agree on this. I also believe that identifier binding is needed in the 'not obvious' cases such as when the identifier is an X.509 DN. When the identifier is a public key then it is obvious how to authenticate the holder of the private key as we have many existing standards that do this i.e. send a challenge and get it signed by the private key. So we do not need identifier bindings for the obvious cases that are implicit in the nature of the identifier.

**Comment by OR13** [comment]
As a note, given the v2 context includes a default vocab... 

We should refer to the "issuing" party of a "verifiable presentation", as an "issuer" since that is what the default vocab will expand terms to be rooted under, see:

https://github.com/w3c/vc-data-model/blob/main/contexts/credentials/v2#L4

^ TLDR: this line currently applies to both VC and VP.

**Comment by OR13** [comment]
I suggest we close this issue, and not define another role for "holder".

**Comment by dlongley** [comment]
> We should refer to the "issuing" party of a "verifiable presentation", as an "issuer" since that is what the default vocab will expand terms to be rooted under, see:

If this is a problem, we could change the URL to be "author-dependent" instead -- which may be more agnostic as to any particular role other than being the author of the data that used the terms that resolved via `@vocab`.

**Comment by awoie** [comment]
@RieksJ do you still strongly believe that we need to have a separation between prover and issuee (both being specializations of holder)? Discussions regarding holder binding or identifier binding are happening in another PR #1054. 

Otherwise I don't see strong consensus in this thread to add that property. Especially since @OR13 who created this issue wanted to close it.

**Comment by dlongley** [comment]
@OR13,

> or "controller-dependent" to align with how "controller" is used.

That's an option, but I'd be worried that we'd hit some similar situation like we just did with "issuer-dependent" where it didn't apply. If we use "author-dependent" then it should be a safe bet because it's always up to the author of the terms to decide what they meant.

**Comment by awoie** [comment]
@dlongley @OR13 Is my understanding correct that we won't have to change holder to prover or introduce a new role prover but we might need to adjust the vocab? If this is correct, what concrete changes have to happen in the vocab and can we create a separate issue for that since this issue is about introducing a new role prover?

**Comment by OR13** [comment]
We don't need to change things, we do need to define holder a bit better.

I think we can close.

**Comment by decentralgabe** [comment]
Is there an issue to track clearly defining `holder`? If not, @OR13 would you want to close this and open that one?
I see @awoie is assigned. Perhaps the holder definition is encompassed in the binding work?

**Comment by Sakurann** [comment]
good discussion - please open a new issue if there is a need to redefine a term holder - discussion in this issue is impossible to follow with that regard.

**Comment by RieksJ** [comment]
> We don't need to change things, we do need to define holder a bit better.

That's not going to help, unless people start using terms in the meanings in which they are defined, which is currently not the case. I contend that if they were to do that, even if the definition weren't modified, that would already lead to the resolution of many issues.

**Comment by RieksJ** [comment]
> Is there an issue to track clearly defining `holder`? If not, @OR13 would you want to close this and open that one? I see @awoie is assigned. Perhaps the holder definition is encompassed in the binding work?

There is a definition of 'holder' in the [RWOT paper on identifier/holder binding](https://github.com/WebOfTrustInfo/rwot11-the-hague/blob/master/final-documents/identifier-binding.pdf), which I think is quite good. But there is no point in having a good definition unless people start to actually commit to using the term in the way it is defined.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-06-21-vcwg#section2-4) on 2023-06-21

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.4. Define "prover" in addition to "holder". (issue vc-data-model#902)

_See github issue [vc-data-model#902](https://github.com/w3c/vc-data-model/issues/902)._




**Kristina Yasuda:** Orie says we can close. I'm putting pending close.  

</details>

**Comment by TallTed** [comment]
@RieksJ — I believe this is the definition of `holder` to which you refer:
> A role that a **party[5]** can perform as it (a) requests and obtains a **VC** from an **issuer**, (b) manages **VCs** within a **wallet**, or (c) creates **VPs** from one or more **VCs** and sends them to the **verifier** that requested it. A **holder** is usually, but not always, the **subject** of a **claim** in one or more of the **VCs** that it uses to create a VP.
>
> [5] The W3C VCDM defines the holder as an ‘**entity**’, leaving it to the reader to determine from the context whether or not a **party** is intended, or a **component** (an **actor**) that acts on behalf of such a **party**. For a discussion about distinguishing between **parties** and **actors**, see the eSSIF-Lab mental model on [Parties, Actors and Actions](https://essif-lab.github.io/framework/docs/terms/pattern-party-actor-action).

If we're going to discuss it here (or in a new issue), I think it makes sense to have it here (or in that new issue).

**Comment by brentzundel** [comment]
No objections raised to closing since marked `pending close`, closing.

I invite those who wish to continue discussing whether to revise the definition of holder to open an issue for that purpose.
